fuel-vm/src/lib.rs

```diff
@@ -2,8 +2,15 @@
 
 #![warn(missing_docs)]
 #![deny(unsafe_code)]
-#![deny(unused_crate_dependencies)]
+// #![deny(unused_crate_dependencies)]
```

Comment:
Can we uncomment it?=)

Reply:
The lint is broken when compiling without `std` feature, and we'd have to work around it quite a bit. Currently it's since some `dev-dependencies` are only used when `feature = "std"`.

The lint itself is quite problematic, see https://github.com/rust-lang/rust/issues/57274 for discussion on that.

Our options here are:
* Removing the lint and moving back to  `cargo-udeps`
* Removing the lint and not automatically checking for unused deps
* Feature-gating the lint: `#[cfg_attr(feature = "std", deny(unused_crate_dependencies))]`
* Or doing this:

```rust
#[cfg(all(test, not(feature = "std")))] use futures as _;
#[cfg(all(test, not(feature = "std")))] use tokio as _;
#[cfg(all(test, not(feature = "std")))] use tokio_rayon as _;
```